subsys/usb/host/class/usbh_uvc.c

```diff
@@ -308,104 +746,2771 @@ static int usbh_uvc_probe(struct usbh_class_data *const c_data,
+
+	return 0;
+}
+
+/**
+ * @brief Enumerate supported frame intervals
+ */
+static int usbh_uvc_enum_frmival(const struct device *dev,
+					    struct video_frmival_enum *frmival_enum)
+{
+	struct uvc_device *uvc_dev = dev->data;
+	int ret;
+
+	if (!uvc_dev->connected || !uvc_dev) {
+		return -ENODEV;
+	}
+
+	k_mutex_lock(&uvc_dev->lock, K_FOREVER);
+
+	ret = uvc_host_enum_frame_intervals(uvc_dev->current_format.frame_ptr, frmival_enum);
+	if (ret != 0) {
+		LOG_DBG("Failed to enumerate frame intervals: %d", ret);
+	}
+
+	k_mutex_unlock(&uvc_dev->lock);
+	return ret;
+}
+
+#ifdef CONFIG_POLL
+/**
+ * @brief Set poll signal for UVC device events
+ */
+static int usbh_uvc_set_signal(const struct device *dev, struct k_poll_signal *sig)
+{
+	struct uvc_device *uvc_dev = dev->data;
+
+	if (!uvc_dev) {
+		LOG_ERR("No UVC device data");
+		return -ENODEV;
+	}
+
+	k_mutex_lock(&uvc_dev->lock, K_FOREVER);
+	uvc_dev->sig = sig;
+	k_mutex_unlock(&uvc_dev->lock);
+
+	LOG_DBG("Signal set for UVC device %p", uvc_dev);
+
+	return 0;
+}
+#endif
+
+/**
+ * @brief Get volatile control values
+ */
+static int usbh_uvc_get_volatile_ctrl(const struct device *dev, uint32_t id)
+{
+	struct uvc_device *uvc_dev = dev->data;
+	struct usb_camera_ctrls *ctrls = &uvc_dev->ctrls;
+	int ret = 0;
+
+	switch (id) {
+	case VIDEO_CID_EXPOSURE_AUTO:
+		/* TODO: Implement exposure value retrieval */
+		break;
+
+	case VIDEO_CID_AUTOGAIN:
+		ret = uvc_host_get_current_gain(uvc_dev, &ctrls->gain.val);
+		if (ret != 0) {
+			LOG_ERR("Failed to get current gain value: %d", ret);
+			return ret;
+		}
+		LOG_DBG("Updated gain to current value: %d", ctrls->gain.val);
+		break;
+
+	case VIDEO_CID_AUTO_WHITE_BALANCE:
+		/* TODO: Implement white balance temperature retrieval */
+		break;
+
+	default:
+		LOG_WRN("Volatile control 0x%08x not supported", id);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Search for control in a specific UVC entity type
+ */
+static int uvc_host_search_control_in_entity(uint8_t subtype, uint32_t cid,
+					      uint8_t *found_type,
+					      const struct uvc_control_map **map)
+{
+	const struct uvc_control_map *map_table;
+	size_t map_length;
+	int ret;
+
+	ret = uvc_get_control_map(subtype, &map_table, &map_length);
+	if (ret != 0) {
+		return ret;
+	}
+
+	for (size_t i = 0; i < map_length; i++) {
+		if (map_table[i].cid == cid) {
+			*found_type = subtype;
+			*map = &map_table[i];
+			return 0;
+		}
+	}
+
+	return -ENOENT;
+}
+
+/**
+ * @brief Find control mapping by CID based on available entities
+ */
+static int uvc_host_find_control_mapping(struct uvc_device *uvc_dev, uint32_t cid,
+					 uint8_t *entity_subtype,
+					 const struct uvc_control_map **map)
+{
+	int ret;
+
+	/* Search in Processing Unit if it exists */
+	if (uvc_dev->uvc_descriptors.vc_pud) {
+		ret = uvc_host_search_control_in_entity(UVC_VC_PROCESSING_UNIT, cid,
+							entity_subtype, map);
+		if (ret == 0) {
```

Comment:
Is it possible to use the conventional `if (err != 0)` handling like described above?